etcdserver/server.go

```diff
@@ -241,6 +250,7 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 		}
 		existingCluster, err := GetClusterFromRemotePeers(getRemotePeerURLs(cl, cfg.Name), prt)
```

Comment:
actually we don't need this I think. What do you mean by another `!haveWAL` case?